pylint/checkers/similar.py

```diff
@@ -26,32 +26,315 @@
+
+    @property
+    def start(self) -> LineNumber:
+        return self._start
+
+    @property
+    def end(self) -> LineNumber:
+        return self._end
+
+    @end.setter
+    def end(self, value: LineNumber):
+        self._end = value
+
+    def __repr__(self) -> str:
+        return f"<SuccessiveLinesLimits <{self._start};{self._end}>>"
+
+
+class LineSetStartCouple(NamedTuple):
+    """
+    Indices in both linesets that mark the beginning of successive lines
+    """
+
+    fst_lineset_index: Index
+    snd_lineset_index: Index
+
+    def __repr__(self) -> str:
+        return (
+            f"<LineSetStartCouple <{self.fst_lineset_index};{self.snd_lineset_index}>>"
+        )
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, LineSetStartCouple):
+            raise NotImplementedError
+        return (
+            self.fst_lineset_index == other.fst_lineset_index
+            and self.snd_lineset_index == other.snd_lineset_index
+        )
+
+    def __hash__(self) -> int:
+        return hash(self.fst_lineset_index) + hash(self.snd_lineset_index)
+
+    def increment(self, value: Index) -> "LineSetStartCouple":
+        return LineSetStartCouple(
+            Index(self.fst_lineset_index + value),
+            Index(self.snd_lineset_index + value),
+        )
+
+
+def hash_lineset(
+    lineset: "LineSet", min_common_lines: int = DEFAULT_MIN_SIMILARITY_LINE
+) -> Tuple[HashToIndex_T, IndexToLines_T]:
+    """
+    Return two dicts. The first associates the hash of successive stripped lines of a lineset
+    to the indices of the starting lines.
+    The second dict, associates the index of the starting line in the lineset's stripped lines to the
+    couple [start, end] lines number in the corresponding file.
+
+    :param lineset: lineset object (i.e the lines in a file)
+    :param min_common_lines: number of successive lines that are used to compute the hash
+    :return: a dict linking hashes to corresponding start index and a dict that links this
+             index to the start and end lines in the file
+    """
+    hash2index = defaultdict(list)
+    index2lines = dict()
+    # Comments, docstring and other specific patterns maybe excluded -> call to stripped_lines
+    # to get only what is desired
+    lines = list(x.text for x in lineset.stripped_lines)
+    # Need different iterators on same lines but each one is shifted 1 from the precedent
+    shifted_lines = [iter(lines[i:]) for i in range(min_common_lines)]
+
+    for index_i, *succ_lines in enumerate(zip(*shifted_lines)):
+        start_linenumber = lineset.stripped_lines[index_i][1]
+        try:
+            end_linenumber = lineset.stripped_lines[index_i + min_common_lines][1]
+        except IndexError:
+            end_linenumber = lineset.stripped_lines[-1][1] + 1
+
+        index = Index(index_i)
+        index2lines[index] = SuccessiveLinesLimits(
+            LineNumber(start_linenumber), LineNumber(end_linenumber)
+        )
```

Comment:
```suggestion
        index2lines[index] = SuccessiveLinesLimits(
            start=LineNumber(start_linenumber),
            end=LineNumber(end_linenumber),
        )
```
That makes it a bit more readable